### PR TITLE
added endpoints for archiving, unarchiving documents

### PR DESCRIPTION
**Pull request by akorneyc**
this pull request includes only endpoints for archiving, unarchiving documents.
also updated endpoint '/documents' with new parameter to filter documents by archiving status

### CODE REVIEW

File: doc/dokuti_api.yml [hunk 1]
```diff
@@ -378,6 +385,108 @@ paths:
...
+              schema:
+                $ref: '#/components/schemas/Document'
+        '400':
+          description: Invalid parameters.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description: Not authorized to update Documents.
```
Comment (gm7) [inline doc/dokuti_api.yml]: not authorized to archive Documents.

File: doc/dokuti_api.yml [hunk 2]
```diff
@@ -14,6 +14,13 @@ paths:
       description: Returns list of all documents associated with the logged in user
       operationId: getDocuments
       parameters:
+        - name: archive
```
Comment (gm7) [inline doc/dokuti_api.yml]: `archive` should be named `filterArchive` to align with other filtering parameters. 
Small concern about checking the user's permissions when setting this parameter to `true`... will be able restrict only users with "archive" permission to see the archived documents?

Comment (VictorSzewczenko) [reply doc/dokuti_api.yml]: @PreAuthorize (Which can only be applied to functions, as far as I know) annotation uses a service under hood to assert that a permission is in the token. You could use this same service to check for permissions within an actual piece of code (like a controller) to determine whether to restrict access to it.

Then you could throw a NotAuthorizedExeption if the permission check fails.

File: doc/dokuti_api.yml [hunk 3]
```diff
@@ -378,6 +385,108 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorResponse'
+  
+  '/documents/{documentId}/archive':
```
[the remark this answers is not in the document]
Comment (melissapalmer) [reply doc/dokuti_api.yml]: This is a similar question to what I asked on previous PR at: 
- https://github.com/GrindrodBank/dokuti/pull/23#discussion_r335318277 
probably some aligning with that `/document` endpoints needs to be looking into.

[the remark this answers is not in the document]
Comment (gm7) [reply doc/dokuti_api.yml]: @VictorSzewczenko I requested a separate endpoint(s) to isolate to the `archive` permission  which, according to our use cases, should be separate from the ability to update a document.

**Pull request merged**